frontend/integration-tests/tests/storage.scenario.ts

```diff
@@ -0,0 +1,62 @@
+import { $$, browser, ExpectedConditions as until, by, element } from 'protractor';
+import { testName, checkLogs, checkErrors } from '../protractor.conf';
+import * as crudView from '../views/crud.view';
+import { execSync } from 'child_process';
+
+describe('Add storage is applicable for all workloads', () => {
+
+  const k8sWorkloads = ['replicationcontrollers', 'daemonsets', 'deployments', 'replicasets', 'statefulsets'];
+  const openshiftWorkloads = ['deploymentconfigs'];
+  const resourceObjs = browser.params.openshift === 'true' ? openshiftWorkloads.concat(k8sWorkloads) : k8sWorkloads;
+
+  afterEach(() => {
+    checkLogs();
+    checkErrors();
+  });
+
+  afterAll(() => {
+    resourceObjs.forEach((resourcetype) => {
+      try {
+        execSync(`kubectl delete --cascade ${resourcetype} example -n ${testName}`);
+      } catch (error) {
+        console.error(`Failed to delete ${resourcetype} example:\n${error}`);
+      }
+    });
+  });
+
+  resourceObjs.forEach((resourcetype) => {
+    describe(resourcetype, () => {
+      const pvcName = `${resourcetype}-pvc`;
+      const pvcSize = '1';
+      const mountPath = '/data';
+      it(`create a ${resourcetype} resource`, async() => {
+        await crudView.createNamespacedResourceWithDefaultYaml(resourcetype);
+        expect(crudView.errorMessage.isPresent()).toBe(false);
+      });
+      it(`add storage to ${resourcetype}`, async() => {
+        if (resourcetype === 'deploymentconfigs' || resourcetype === 'deployments') {
+          const activeStatus = element(by.cssContainingText('.co-icon-and-text', 'Up to date'));
+          while (!activeStatus.isPresent()) {
+            await browser.sleep(10000);
```

Comment:
Does this go into an infinite loop if the deployment never rolls out?

I'm not sure we need to check whether it rolls out successfully. Is it sufficient to check that the volume table has the right values? This is going to make the tests a lot slower, and I'm not sure it has a ton of value.

Reply:
It's not suitable to use while in the test, I updated it with if, but need to make the wait time long enough.
This check is before adding storage. Since the workload is just created in previous "it" test, and sometimes adding storage would meet confliction to edit yaml. So I want to wait after the previous deployment is completed. Do you have better approach for this situation?